### PR TITLE
Extract app() into separate function

### PR DESCRIPTION
**Pull request by seanh**
Extract the code for reading the config.app setting into a separate
function.

This function can now be unit tested separately, and the long
configFrom() function gets a little shorter.

### CODE REVIEW

File: src/annotator/config/settings.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+ * Return the value of the href attribute of the first
+ * `<link type="application/annotator+html">` element in the given document.
+ *
+ * This URL is used as the src of the sidebar's iframe.
+ *
+ * @param {Document} - The document to search.
+ * @return {string} - The URL to use for the sidebar's iframe.
+ *
+ * @throws {Error} - If there's no annotator link or the first annotator has
+ *   no href.
```
Comment (seanh) [inline src/annotator/config/settings.js]: Previously the code would throw this error, but the fact wasn't documented or tested. (Not hugely important - in practice the link is always there.)

File: src/annotator/config/test/index-test.js [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
   jsonConfigsFrom: sinon.stub(),
 };
 var fakeSettings = {};
 
-var configFrom = proxyquire('../index', {
+var configFrom = proxyquire('../index', util.noCallThru({
   './settings': fakeSettings,
   '../../shared/settings': fakeSharedSettings,
-});
+}));
```
Comment (seanh) [inline src/annotator/config/test/index-test.js]: Lack of `noCallThru` here cost me a bunch of time when I had a typo in a test fake

File: src/annotator/config/test/index-test.js [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
 
-function fakeWindow(someHref) {
-  var href = someHref || 'LINK_HREF';
-
+function fakeWindow() {
   return {
-    document: {
-      querySelector: sinon.stub().returns({href: href}),
-    },
+    document: 'THE_DOCUMENT',
```
Comment (seanh) [inline src/annotator/config/test/index-test.js]: In the `index.js` tests we no longer need to mock the DOM (`querySelector` etc). In the `settings.js` unit tests for any particular settings functions that do use the DOM, we can use the real DOM just for those functions own unit tests.

File: src/annotator/config/index.js [hunk 4]
```diff
@@ -12,9 +12,9 @@ var docs = 'https://h.readthedocs.io/en/latest/embedding.html';
  */
 function configFrom(window_) {
   var config = {
-    app: window_.
-      document.querySelector('link[type="application/annotator+html"]').href,
+    app: settings.app(),
```
Comment (sean-roberts) [inline src/annotator/config/index.js]: don't you need to pass the document?

Comment (seanh) [inline src/annotator/config/index.js]: Heh, yes

Comment (seanh) [inline src/annotator/config/index.js]: Fixed, and I added a collaboration test for it

**Comment by robertknight** [comment]
LGTM.

**Pull request merged**